packages/sogrim-app/src/components/Banner/BannerCards/DegreeStatusBar.tsx

```diff
@@ -14,18 +17,87 @@ const LinearProgressWithLabel = (
         <LinearProgress variant="determinate" {...props} color="secondary" />
       </Box>
       <Box sx={{ minWidth: 35, mr: -1 }}>
-        <Typography variant="body2" color="text.secondary">{`${Math.round(
-          props.value
-        )}%`}</Typography>
+        <Typography variant="body2" color="text.secondary">
+          {`${Math.round(props.value)}%`}
+        </Typography>
       </Box>
     </Box>
   );
 };
 
-export const DegreeStatusBar = ({ progress = 0 }: { progress: number }) => {
+const MultiColorLinearProgress = styled(LinearProgress)(({ theme }) => ({
+  [`& .${linearProgressClasses.dashed}`]: {
+    backgroundImage: "none",
+    backgroundColor: "rgba(0, 0, 0, 0.178)",
+    animation: "none",
+  },
+
+  [`& .${linearProgressClasses.bar1Buffer}`]: {
+    backgroundColor: theme.palette.secondary.main,
+  },
+  [`& .${linearProgressClasses.bar2Buffer}`]: {
+    backgroundColor: theme.palette.secondary.light,
+  },
+}));
+
+const LinearProgressMultiColorWithLabel = (
+  props: LinearProgressProps & {
+    onlyCompleteProgress: number;
+    totalProgress: number;
+  }
+) => {
+  const theme = useTheme();
   return (
```

Comment:
@liadaram1  @benny-n 
best practice for tsx file - "only one return". i mean, only one component so we need to break this file.
each file - component.